### DISCUSSION TITLE
[Q:] can the homogenization within new TM system be utilized to plasticity not only elasticity ?

### DISCUSSION THREAD

**Discussion by Ttw0626** [idaholab/moose#25972] [Q&A Modules: Solid mechanics] [answered]
Dear TM-developers,

`Introduction`
**The homogenization of new system is working well in all of the elasticity problem no matter small and finite strain,
But if I changed the material to plasticity, the average cell value (homogenzied) is not correct and only present
the elastic response.**

`Interesting`
**the output of effective Plastic strain is correct.**

`Q`
**could you as experts re-implement the plastic problem under the homogenization of new system ?**

`By the way`
**dose this kind of method for presenting homogenization have bad-effect on the convergence of plastic problem especially in global (after assembly of finite element) ?**

**how to control this method related to the Macro-deformation gradient in finite strain ? especially in plasticity**

Thanks so much

best regards,
tianwen

**Comment by Ttw0626** [comment]
[example.zip](https://github.com/idaholab/moose/files/13276649/example.zip)


here is test example, yeah, the homogeneous case is also sutiable to this homogenization system, I think.


**Important Updating**
I read a kind of report from ANL "Identify the influence of microstructure on mesoscale creep and fatigue damage" in 2020
almost 4 years ago.

one of sentence from it is  **The current MOOSE constraint system is unsuited for implementing Multiple points (dummy nodes) types of conditions in 3D and so we examined other approaches. by Andrea Rovinelli**

perhaps, I got misunderstanding for this homogenization system which should be fine at any material. I will check again.

**Comment by RossMech** [comment]
A question to your issue. Did it work in the end as planned? I would also like to make the creep/viscoplastic homogenization using the new kernel system.

**Reply by Ttw0626** [reply]
Dear Ross,

hi

**answer:**
the implementation of my own material based on the new system was not going well, unfortunately.
The unconverged problem occured at the FEproblem issues, maybe due to the "UserObjects" of homogenization.

By the way, you could try your model (eg. the viscoplasticity). This homogenization theory was not based on boundary-integral implemented by multiple-point-constraints method common in commercial software, like **abaqus Equation**. though they are similar from theoretical part, the implementation has some problem for me.

if you wanna focus on this point, see this report ANL "Identify the influence of microstructure on mesoscale creep and fatigue damage" in 2020.

Thanks
kind regards,
tianwen

**Reply by RossMech** [reply]
Dear Tianwen,

thank you very much for your response. I will look into the report from ANL you mentioned.

Best regards,
Rostyslav.

**Comment by RossMech** [comment]
I had the same problem with the standard creep model. If the homogeneous material properties are given into the homogenization framework, the system's output converges to the right value. However, if creep parameters vary across different blocks, the output diverges.

**Comment by Ttw0626** [comment, marked as answer]
Hi, moose-developers,

sorry this borthering. my issues about this had already solved personally due to the problem of my code.

this discussion will be closed.

Thanks
tianwen
